tableauserverclient/server/endpoint/schedules_endpoint.py

```diff
@@ -58,3 +63,22 @@ def create(self, schedule_item):
         new_schedule = ScheduleItem.from_response(server_response.content, self.parent_srv.namespace)[0]
         logger.info("Created new schedule (ID: {})".format(new_schedule.id))
         return new_schedule
+
+    @api(version="2.8")
+    @item_must_be_of_type(workbook_or_datasource=[WorkbookItem, DatasourceItem])
+    def add_to_schedule(self, schedule_id, workbook_or_datasource):
+        type_ = item_type(workbook_or_datasource)
+
+        # id will exist because item_must_be_of_type ensures this
+        id_ = workbook_or_datasource.id
+        req_factory = getattr(RequestFactory.Schedule, 'add_{}_req'.format(type_), None)
```

Comment:
Similar comments to above -- I think this can just be a simple if statement on the type (or a mapping if there are lots of types) but that feels more robust than the meta-programming with getattr